src/components/SingleArticle.jsx

```diff
@@ -1,37 +1,74 @@
 import { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
-import { getArticleById, getCommentsByArticleId } from "../../utils/api";
+import { getArticleById, getCommentsByArticleId, patchArticle } from "../../utils/api";
 import CommentCard from "./CommentCard";
 
 function SingleArticle() {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
   const [comments, setComments] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
+  const [articleLoading, setArticleLoading] = useState(true);
+  const [commentsLoading, setCommentsLoading] = useState(true);
+  const [voteError, setVoteError] = useState(null);
+
+  function upVote(article_id) {
+    const body = {inc_votes: 1}
+    setArticle((currArticle) => {
+      return {...currArticle, votes: currArticle.votes + 1}
+    })
+    setVoteError(null);
+    patchArticle(article_id, body).catch((err)=>{
+      setArticle((currArticle) => {
+        return {...currArticle, votes: currArticle.votes - 1}
+      })
+      setVoteError('Something went wrong, please try again')
```

Comment:
Great that you've considered the error handling here - but it would be better if this was a little text message that popped up next to the vote buttons instead of getting rid of the whole article page in favor of just showing the error.